services/github/github-constellation.js

```diff
@@ -70,8 +80,21 @@ class GithubConstellation {
       log.error(e)
     }
 
+    // Reserve a subset of scoped tokens from the total set
+    // to be used for queries which require an explicit scope,
+    // while leaving a sufficient amount of tokens (scoped or unscoped)
+    // for the bulk of our requests which don't care about scopes.
+    this._maxNumReservedScopedTokens = Math.floor(tokens.length * 0.15)
```

Comment:
If we opt to go with the approach proposed in this PR, then I could envision a point down the line where we allow this ratio to be set from an environment variable so that we can adjust via a simple heroku config update vs code change/PR